packages/klevu-ui/src/components/klevu-popup/klevu-popup.tsx

```diff
@@ -205,12 +207,41 @@ export class KlevuPopup {
     }
   }
 
-  connectedCallback() {
-    if (this.closeAtOutsideClick) {
+  #backdropClick(event: MouseEvent) {
+    if (!this.dialogRef || !this.dialogRef.open) {
+      return
+    }
+
+    const rect = this.dialogRef.getBoundingClientRect()
+    const isInDialog =
+      rect.top <= event.clientY &&
+      event.clientY <= rect.top + rect.height &&
+      rect.left <= event.clientX &&
+      event.clientX <= rect.left + rect.width
```

Comment:
just checking the need to do this, we can do this by checking `!this.dialogRef.contains(event.target)`, something along these lines where we check if the event originated from the modal or outside.
did you already try this approach?

Reply:
Dialog is a bit of exception. You will always get the dialog as source of the event so you have to check that is inside actual dialog or in the backdrop.